4-reduction/scipp-introduction.ipynb

```diff
@@ -1259,7 +1260,231 @@
+   "source": [
+    "signal.hist(x=200, y=200).plot() + signal.hist(x=4, y=4).plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "22acde22-4116-4a5e-8a24-d24e35acde2f",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "**Solution:**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9a749c4c-aae8-4d39-92fc-9d81fc113ade",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": [
+     "hide-cell",
+     "solution"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# 30~50 bins are enough to see the meaningful shape!\n",
+    "signal.hist(x=50, y=50).plot() + signal.hist(x=30, y=30).plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "596a2c0a-c653-4ccd-9797-0232642cde56",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "#### Exercise 3-2: Custom histogram edges.\n",
+    "\n",
+    "However, there is a suspicious hot spot in the very middle of the image.\n",
+    "\n",
+    "We want to investigate those signals within the specific range of ``x`` and ``y``.\n",
+    "\n",
+    "Let's histogram the hot spot and see what is in there.\n",
+    "\n",
+    "You can histogram the data with custom histogram edges like below."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f448c97d-91da-4ee3-bee4-8c8b87d0ffb9",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "**Hint:**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8c017f32-9082-4fcf-b7da-9ede914d397b",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": [
+     "remove-output"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "hist_edges_x = sc.linspace(dim='x', start=-10, stop=10, unit='cm', num=200)\n",
+    "hist_edges_y = sc.linspace(dim='y', start=-10, stop=10, unit='cm', num=200)\n",
+    "signal.hist(x=hist_edges_x, y=hist_edges_y).plot()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a2f5d401-7d07-416c-90fb-d39185862431",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "**Solution:**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8bbb4e50-614d-49d3-a873-1493b45cb9bd",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": [
+     "solution",
+     "hide-cell"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# There was a smiley in the middle of the heart!\n",
+    "\n",
+    "hist_edges_x = sc.linspace(dim='x', start=-0.1, stop=0.1, unit='cm', num=50)\n",
+    "hist_edges_y = sc.linspace(dim='y', start=-0.1, stop=0.1, unit='cm', num=50)\n",
```

Comment:
```suggestion
    "hist_edges_x = sc.linspace(dim='x', start=-0.15, stop=0.15, unit='cm', num=50)\n",
    "hist_edges_y = sc.linspace(dim='y', start=-0.15, stop=0.15, unit='cm', num=50)\n",
```

I think it looks even better with a bit of padding around it?
![Screenshot at 2023-09-01 14-10-34](https://github.com/ess-dmsc-dram/dmsc-school/assets/39047984/ae1b167b-be32-4b28-b543-1b61d892b184)